hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -670,18 +685,25 @@ public void setIoException(Exception e) {
   void cleanup() {
   }
 
-  public void cleanup(boolean invalidateClient) {
+  public void cleanup(boolean invalidateClient) throws IOException {
     if (xceiverClientFactory != null) {
       xceiverClientFactory.releaseClient(xceiverClient, invalidateClient);
     }
     xceiverClientFactory = null;
     xceiverClient = null;
     cleanup();
 
-    if (bufferList !=  null) {
+    if (bufferList != null) {
       bufferList.clear();
     }
     bufferList = null;
+    if (lastChunkBuffer != null) {
+      try {
+        cleanDirectByteBuffer();
```

Comment:
Add `isDirect()` check before calling this one? And that `Preconditions.checkArgument()` can be removed.

```suggestion
    if (lastChunkBuffer != null && lastChunkBuffer.isDirect()) {
      try {
        cleanDirectByteBuffer();
```